driver-core/src/test/unit/com/mongodb/internal/session/ServerSessionPoolSpecification.groovy

```diff
@@ -253,35 +207,22 @@ class ServerSessionPoolSpecification extends Specification {
             getCurrentDescription() >> connectedDescription
         }
         def pool = new ServerSessionPool(cluster, getServerApi())
-        // check out sessions up to the endSessions batch size
         def sessions = []
-        10000.times { sessions.add(pool.get()) }
-        // and then check out one more
-        def oneOverBatchSizeSession = pool.get()
+        10.times { sessions.add(pool.get()) }
 
-        // now release them all before closing the pool
         for (def cur : sessions) {
             pool.release(cur)
         }
-        pool.release(oneOverBatchSizeSession)
 
         when:
         pool.close()
 
         then:
-        // first batch is the first 10K sessions, final batch is the last one
```

Comment:
The whole batch size thing is silly.  10K sessions doesn't even come close to reaching 16M command document size limit.  And it's a SHOULD in the spec, so I think it's ok to rip it out.